### ISSUE TITLE
Exception when enabling unity build

### ISSUE THREAD

**Issue by TingPing** [mesonbuild/meson#923]
```python
Traceback (most recent call last):
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/mesonmain.py", line 276, in run
    app.generate()
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/mesonmain.py", line 169, in generate
    g.generate(intr)
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/backend/ninjabackend.py", line 187, in generate
    self.generate_target(t, outfile)
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/backend/ninjabackend.py", line 255, in generate_target
    self.process_target_dependencies(target, outfile)
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/backend/ninjabackend.py", line 364, in process_target_dependencies
    self.generate_target(t, outfile)
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/backend/ninjabackend.py", line 255, in generate_target
    self.process_target_dependencies(target, outfile)
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/backend/ninjabackend.py", line 364, in process_target_dependencies
    self.generate_target(t, outfile)
  File "/home/tingping/.local/lib/python3.5/site-packages/meson-0.36.0.dev1-py3.5.egg/mesonbuild/backend/ninjabackend.py", line 286, in generate_target
    unity_deps.append(os.path.join(self.environment.get_build_dir(), RawFilename(src)))
  File "/usr/lib/python3.5/posixpath.py", line 89, in join
    genericpath._check_arg_types('join', a, *p)
  File "/usr/lib/python3.5/genericpath.py", line 143, in _check_arg_types
    (funcname, s.__class__.__name__)) from None
TypeError: join() argument must be str or bytes, not 'RawFilename'
```

**Comment by nirbheek** [comment]
I fixed this in #908.

**Comment by nirbheek** [comment]
This is fixed now.
